SoptAssignment1/app/src/main/java/com/example/soptassignment1/SignUpActivity.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.soptassignment1
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import com.example.soptassignment1.databinding.ActivitySignUpBinding
+
+
+class SignUpActivity : AppCompatActivity() {
+    private lateinit var binding : ActivitySignUpBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // 로그인 뷰로 이동
+        binding.btnSignup.setOnClickListener {
+            if(binding.etName.text.isNullOrBlank() || binding.etID.text.isNullOrBlank() || binding.etPassword.text.isNullOrBlank())
+                Toast.makeText(this, "입력되지 않은 정보가 있습니다", Toast.LENGTH_SHORT).show()
+            else {
+                val intent = Intent(this, SignInActivity::class.java)
+                intent.putExtra("ID", binding.etID.text.toString())
+                intent.putExtra("password", binding.etPassword.text.toString())
+                setResult(RESULT_OK, intent)
```

Comment:
두 번째로.. 조금 어려운 방식을 얘기하자면.. 코틀린의 스코프 함수(apply, with, let, also, run)라는 것을 이용해서(저는 이 중 apply를 사용하겠습니다~)

```suggestion
           setResult(RESULT_OK, Intent(this, SignInActivity::class.java).apply {
                      putExtra("ID", binding.etID.text.toString())
                      putExtra("password", binding.etPassword.text.toString())
           })
```
이렇게도 표현이 가능합니다~!! 스코프 함수의 경우 좀 더 자세히 정리해놓은 [블로그 링크](https://medium.com/@limgyumin/코틀린-의-apply-with-let-also-run-은-언제-사용하는가-4a517292df29)를 두고 갈테니 확인해보시길 바라요~👀

Reply:
확실히 코드가 줄어드네요. 좀 더 공부하겠습니다